megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -1948,30 +1948,27 @@ FiringPlan getBestFiringPlanWithTwists(Entity shooter, Targetable target,
             return noTwistPlan;
         }
 
-        // Turn to the right.
-        shooter.setSecondaryFacing(correctFacing(originalFacing + 1));
-        FiringPlan rightTwistPlan = getBestFiringPlan(shooter, target, game, ammoConservation);
-        rightTwistPlan.setTwist(1);
-
-        // Turn to the left.
-        shooter.setSecondaryFacing(correctFacing(originalFacing - 1));
-        FiringPlan leftTwistPlan = getBestFiringPlan(shooter, target, game, ammoConservation);
-        leftTwistPlan.setTwist(-1);
-
-        // todo extended torso twist.
-
+        ArrayList<Integer> validFacingChanges = getValidFacingChanges(shooter);
+        
+        // Now, we loop through all possible facings. If one facing produces a better plan 
+        // than what we currently have as the best plan then use that. Start with "no twist" as default.
+        FiringPlan bestFiringPlan = noTwistPlan;
+        for(int index = 0; index < validFacingChanges.size(); index++)
+        {
+        	int currentTwist = validFacingChanges.get(index);
+        	
+        	shooter.setSecondaryFacing(correctFacing(originalFacing + currentTwist));
+        	FiringPlan twistPlan = getBestFiringPlan(shooter, target, game, ammoConservation);
+        	twistPlan.setTwist(currentTwist);
+        	
+        	if(twistPlan.getUtility() > bestFiringPlan.getUtility())
+        		bestFiringPlan = twistPlan;
```

Comment:
Put brackets around your clauses.  It makes it easier to extend the clause later if need be.